code/script.js

```diff
@@ -1,325 +1,356 @@
+    // Shows the correct alert message for different categories
+    if (category === 'hair') {
+      if (keep) {
+        alert(`Yes, the person has ${value} hair! Keep all people that has ${value} hair!`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+      } else {
+        alert(`No, the person doesn't have ${value} hair! Remove all people that has ${value} hair!`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      }
+    } else if (category === 'eyes') {
+      if (keep) {
+        alert( `Yes, the person has ${value} eyes! Keep all people that has ${value} eyes!`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+      } else {
+        alert(`No, the person doesn't have ${value} eyes! Remove all people that has ${value} eyes!`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      }
+    } else if (category === 'accessories') {
+      if (keep) {
+        alert(`Yes, the person wears ${value}! Keep all people that wears ${value}`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+      } else {
+        alert(`No, the person does not wear ${value}! Keep all people that does not wear ${value}`)
+        charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+      }
+    } else if (category === 'other') {
+      if (keep) {
+        alert(`Yes, the person is a ${value}! Keep all people that is a ${value}!`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+      } else {
+        alert(`No, the person is not a ${value}! Keep all people that is a ${value}!`)
+        charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+      }
+    }
+    //Redraw the game board.
+    generateBoard()
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
+    //When clicking guess, the player has to confirm that they want to make a guess.
+    const guess = (personToConfirm) => {
+    let confirmPerson = confirm(`Do you really want to make a guess on ${personToConfirm}?`);
+    if (confirmPerson === true) {
+      checkMyGuess (personToConfirm)
+    } else {
+      alert('Ok, no problem! Keep on guessing!')
```

Comment:
I like the detail of adding an else message if the player presses cancel, it's good interaction!